build.xml

```diff
@@ -525,7 +521,34 @@ processor=x86-64;osname=openbsd,
 com/sun/jna/darwin/libjnidispatch.jnilib;
 osname=macosx;processor=x86;processor=x86-64;processor=ppc
 "/>
-      </manifest>
+          </manifest>
+          <manifest file="@{target}" mode="update" if:true="@{module-info}">
+              <attribute name="Multi-Release" value="true"/>
+          </manifest>
+          <manifest file="@{target}" mode="update" unless:true="@{module-info}">
+              <attribute name="Automatic-Module-Name" value="com.sun.jna"/>
+          </manifest>
+      </sequential>
+  </macrodef>
+    <delete dir="${build}/manifest"/>
+    <mkdir dir="${build}/manifest"/>
+    <build-manifest target="${build}/manifest/module.mf" module-info="true"/>
+    <build-manifest target="${build}/manifest/automatic.mf" module-info="false"/>
+    <ModuleGenerator
+        targetFile="${build}/manifest/module-info.class"
+        name="com.sun.jna"
+        version="${jna.major}.${jna.minor}.${jna.revision}"
+        open="false"
+    >
+        <exports package="com.sun.jna" />
+        <exports package="com.sun.jna.ptr" />
+        <exports package="com.sun.jna.win32" />
+        <exports package="com.sun.jna.internal" to="com.sun.jna.platform"/>
+        <requires module="java.logging" />
+        <requires module="java.desktop" />
```

Comment:
BTW, does `jna.jar` actually require `java.desktop` module? Couldn't find which class(es) references it 🤔

Reply:
Don't see anything in the main classes but there are a few test classes which load `java.awt.*` or Swing components.

Reply:
`com.sun.jna.Native`:

```java
import java.awt.Component;
import java.awt.GraphicsEnvironment;
import java.awt.HeadlessException;
import java.awt.Window;
```

Going back 10+ year I would vote to separate that out into its own package, regrettably I don't own a time machine :-)
If I remember correctly I had hoped, that we could get away with `requires transitive`, but it failed at load time, so the higher `require` requires was chosen. The primary intent of the artifacts is, that it is possible to use JNA on the module path, not more, not less.